e2e_test/source/basic/alter/kafka.slt

```diff
@@ -0,0 +1,93 @@
+statement ok
+CREATE SOURCE s1 (v1 int) with (
+  connector = 'kafka',
+  topic = 'kafka_alter',
+  properties.bootstrap.server = 'message_queue:29092',
+  scan.startup.mode = 'earliest'
+) FORMAT PLAIN ENCODE JSON;
+
+statement ok
+CREATE SOURCE s2 (v2 varchar) with (
+  connector = 'kafka',
+  topic = 'kafka_alter',
+  properties.bootstrap.server = 'message_queue:29092',
+  scan.startup.mode = 'earliest'
+) FORMAT PLAIN ENCODE JSON;
+
+statement ok
+create materialized view mv1 as select * from s1;
+
+statement ok
+create materialized view mv2 as select * from s2;
+
+sleep 10s
+
+statement ok
+flush;
+
+query I rowsort
+select * from s1;
+----
+1
+2
+
+query T rowsort
+select * from s2;
+----
+11
+22
+
+# alter source
+statement ok
+alter source s1 add column v2 varchar;
+
+# alter source with null column
+statement ok
+alter source s2 add column v3 int;
```

Comment:
Would you please also test that
- the definition after `alter`
- the next `alter` still works after one `alter`